### PR TITLE
PythonAnalyzer improvements

### PR DESCRIPTION
**Pull request by cmanto**
Running the analyzer against astropy showed an important number of warnings as indicated on #1549

This patch Python-specific issues by implementing "last wins" semantics and better function-local class handling.

Changes:
- Implement Python-specific behavior: skip property setters, transform function-local class chains to use $ scoping, and replace prior duplicates (module and nested) with the latest definitions.
- Change TreeSitterAnalyzer.buildParentFqName signature to accept CodeUnit; add shouldSkipNode, addTopLevelCodeUnit, addChildCodeUnit and recursive removal helpers.
- Introduce CaptureNames with canonical capture-string constants and replace in C#, C++, Go, Java, JS, PHP, Python, Rust, Scala, TS analyzers.
- Add extensive unit tests and Python test fixtures to verify local-class scoping, duplicate field/method handling, and property filtering. Small test runner and minor imports adjusted.

With this change running on astropy remove all errors and warnings on the python side

### CODE REVIEW

**Comment by jbellis** [comment]
unfortunate timing b/c I just extracted and committed 6640410e5a6815ae0d4fbfc41657191e12338bd2 from PR 1501

totally fine to revert that if this has a better solution

(have not looked closely at this but it looks like there may be some overlap)

File: app/src/main/java/io/github/jbellis/brokk/analyzer/TreeSitterAnalyzer.java [hunk 1]
```diff
@@ -1099,6 +1084,118 @@ protected String getLanguageSpecificIndent() {
...
+            Map<CodeUnit, List<String>> localSignatures,
+            Map<CodeUnit, List<Range>> localSourceRanges,
+            ProjectFile file) {
+
+        boolean alreadyExists =
+                localTopLevelCUs.stream().anyMatch(existing -> existing.fqName().equals(cu.fqName()));
+
+        if (!alreadyExists) {
+            localTopLevelCUs.add(cu);
+        } else if (language == Languages.PYTHON && (cu.isField() || cu.isClass() || cu.isFunction())) {
```
[the remark this answers is not in the document]
Comment (cmanto) [reply app/src/main/java/io/github/jbellis/brokk/analyzer/TreeSitterAnalyzer.java]: you're right. I don't like it too much either, but we have some other cases for typescript. 
Perhaps we should aim to remove all of them and make them into overrides?
